src/gui/torrentcontentmodel.cpp

```diff
@@ -497,11 +497,45 @@ void TorrentContentModel::setupModelData(const BitTorrent::AbstractFileStorage &
     qDebug("Torrent contains %d files", filesCount);
     m_filesIndex.reserve(filesCount);
 
-    TorrentContentModelFolder *currentParent;
+    class FolderAdaptor
+    {
+    public:
+        using FolderAdaptorPtr = std::shared_ptr<FolderAdaptor>;
+
+        explicit FolderAdaptor(TorrentContentModelFolder *folder)
+            : m_folder {folder}
+        {
+        }
+
+        FolderAdaptorPtr findChildFolderOrInsert(const QString &folderName)
+        {
+            FolderAdaptorPtr folder = m_folders.value(folderName);
+            if (folder)
+                return folder;
+            folder = std::make_shared<FolderAdaptor>(new TorrentContentModelFolder(folderName, m_folder));
+            m_folder->appendChild(folder->m_folder);
+            m_folders.insert(folderName, folder);
+            return folder;
+        }
+
+        TorrentContentModelFile *insertFile(const QString &fileName, const qulonglong fileSize, const int fileIndex)
```

Comment:
`qlonglong`

Reply:
TorrentModelFile takes qulonglong

Reply:
Then `TorrentModelFile` needs update. The usage site of this function pass in `qlonglong`, isn't? Just stick with `qlonglong`.